frontend/src/pages/Register/RegisterForm.jsx

```diff
@@ -9,95 +14,200 @@ import styles from './Register.module.css';
+    special: true,
+    length: true,
+  });
+
+  const [passwordMatchError, setPasswordMatchError] = useState('');
+  const [passwordListVisible, setPasswordListVisible] = useState(false);
+
+  // State variable to manage the disabled state
+  const [isDisabled, setIsDisabled] = useState(true);
+
+  // Function to toggle the disabled state
+  const toggleDisabled = () => {
+    setIsDisabled(!isDisabled);
+  };
+  // Effect hook to handle errors received from the backend
   useEffect(() => {
-    if (props.paramEmail) {
-      setEmail(props.paramEmail);
+    if (props.errorMessage) {
+      setErrorMessage(props.errorMessage);
     }
-  }, [props.paramEmail]);
-
-  const handleEmailChange = ({ target: { value } }) => setEmail(value);
-  const handlePasswordChange = ({ target: { value } }) => setPassword(value);
-  const handlePasswordConfirmChange = ({ target: { value } }) =>
-    setPasswordConfirm(value);
-
-  const handleFormChange = () => !isTouched && setIsTouched(true);
-
-  const handleSubmit = (event) => {
-    event.preventDefault();
-
-    const form = event.currentTarget;
-
-    if (form.checkValidity()) {
-      props.onSubmitData(email, password).catch((e) => {
-        setErrorMessage(e.response.data.message);
-      });
+  },[props.errorMessage]);
+  // Event handler for email change
+  const handleEmailChange = (e) => {
+    setEmail(e.target.value);
+  };
+  // Event handler for password change
+  const handlePasswordChange = (e) => {
+    setPassword(e.target.value);
+    validatePassword(e.target.value);
+  };
+  // Event handler for confirming password change
+  const handleConfirmPasswordChange = (e) => {
+    setConfirmPassword(e.target.value);
+    validateConfirmPassword(e.target.value);
+  };
+  // Event handler for focusing on the password field
+  const handlePasswordFocus = () => {
+    setPasswordListVisible(true);
+    setPasswordMatchError('');
+  };
+  // Event handler for focusing on the confirm password field
+  const handleConfirmPasswordFocus = () => {
+    setPasswordListVisible(false);
+    setPasswordMatchError('');
+  };
+  // Event handler for handling email blur
+  const handleEmailBlur = () => {
+    if (!validateEmail(email)) {
+      setEmailError('Please enter a valid email address.');
     } else {
-      setErrorMessage(DEFAULT_ERROR_MESSAGE);
+      setEmailError('');
+      toggleDisabled();
     }
+  };
+    // Function to validate email format
+  const validateEmail = (emailValue) => {
+    return /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(emailValue);
+  };
+    // Function to validate all password errors
+  const allPasswordErrorsChecked = Object.values(passwordErrors).every((error) => !error);
+  // Function to validate password format
+  const validatePassword = (passwordValue) => {
+    const regexUpperCase = /[A-Z]/;
+    const regexLowerCase = /[a-z]/;
+    const regexNumber = /\d/;
+    const regexSpecialChar = /[!@#$%^&*()_+=[\]{};':"\\|,.<>?-]/;
+    const regexLength = /^.{8,25}$/;
+
```

Comment:
It is better to take these constants out of the react component.
This way they will only be announced once.
Currently, they will be created every time this function is called.